### PR TITLE
Fix plot resize handling

### LINKED ISSUE #3033: [Plots] Should not make requests every time a user clicks on the plot
Plots should not be requesting new data every time a user clicks on the plot. It should only re-request when the user uses marquee zoom or pan.

## Steps to Reproduce
1. Use console to set a breakpoint inside of `request()` in `TelemetryAPI.js`
1. Open a plot view
1. Observe that breakpoint is hit
1. Resume app
1. Zoom or pan and observe that breakpoint is hit again
1. Resume app
1. click and observe that breakpoint is hit again

## Behavior
**Expected** - Request not called on click (Breakpoint is not hit on click)
**Actual** - Request is called on click

### PR DESCRIPTION
**Pull request by shefalijoshi**
Closes #3033 #5395 VIPEROMCT-116

### Describe your changes:
Don't request data if mouse click has no action
Don't request data if resize width is too small.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Review by unlikelyzero — CHANGES_REQUESTED** [review]
Let's add a regression test for this since it's re-regressed at least three times since I started

**Review by akhenry — CHANGES_REQUESTED** [review]
Looks good, I think this is going to fix this issue once and for all :) Thanks for taking a look at it.

A couple of comments inline.

File: src/plugins/plot/MctPlot.vue [hunk 1]
```diff
@@ -212,6 +212,8 @@ import XAxis from "./axis/XAxis.vue";
 import YAxis from "./axis/YAxis.vue";
 import _ from "lodash";
 
+const OFFSET_THRESHOLD = 50;
```
[the remark this answers is not in the document]
Comment (shefalijoshi) [reply src/plugins/plot/MctPlot.vue:215]: Chrome sends out 2 resize events sometimes and when we switch tabs there is a very small jump in size and then it goes back. So the threshold tries to alleviate unnecessary resizing.

File: src/plugins/plot/MctPlot.vue [hunk 2]
```diff
@@ -1109,8 +1113,10 @@ export default {
             this.freeze();
         },
 
-        play() {
+        play(skipReloadOnInteraction) {
+            this.skipReloadOnInteraction = skipReloadOnInteraction === true;
             this.clear();
```
[the remark this answers is not in the document]
Comment (shefalijoshi) [reply src/plugins/plot/MctPlot.vue:1118]: Yes, that's correct.

File: src/plugins/plot/MctPlot.vue [hunk 3]
```diff
@@ -1109,8 +1113,10 @@ export default {
             this.freeze();
         },
 
-        play() {
+        play(skipReloadOnInteraction) {
```
[the remark this answers is not in the document]
Comment (shefalijoshi) [reply src/plugins/plot/MctPlot.vue]: I agree. I've refactored the code, removing the need for the interaction flag. 
Also renamed some methods to clarify intention.

File: src/plugins/plot/MctPlot.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (akhenry) [reply src/plugins/plot/MctPlot.vue:215]: I guess I'm curious whether a smaller number might work here. Or whether we should be debouncing something that we're not right now to remove noise?

File: src/plugins/plot/MctPlot.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (akhenry) [reply src/plugins/plot/MctPlot.vue]: Changes look great, thank you!

**Comment by akhenry** [comment]
@shefalijoshi Changes look really awesome, it's much cleaner now. Just wondering whether we can get away with a smaller threshold number than 50. Did it not work with smaller numbers than that?

File: src/plugins/plot/MctPlot.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (shefalijoshi) [reply src/plugins/plot/MctPlot.vue:215]: The call is already debounced - I didn't see any improvements to make there.
As for the number - We could go with a lower number, but I saw 2 resize calls - One for a 7 px resize and one for 30something, so I went with 50.

File: src/plugins/plot/MctPlot.vue [hunk 4]
```diff
@@ -1195,7 +1197,8 @@ export default {
             //we ignore when width gets smaller
             const offsetChange = newOffsetWidth - this.offsetWidth;
             if (this.$parent.$refs.plotWrapper
-                && offsetChange >= OFFSET_THRESHOLD) {
+                && offsetChange > OFFSET_THRESHOLD) {
+                console.log('inside');
```
Comment (akhenry) [inline src/plugins/plot/MctPlot.vue:1201]: Debugging code, let's remove.

**Review by akhenry — CHANGES_REQUESTED** [review]
Just one debugging statement left in there, otherwise looks great!

**Comment by ozyx** [comment]
This will need to be cherry picked onto 2.1.0 before merging, looks like some changes from master are tagging along.

**Comment by ozyx** [comment]
> This will need to be cherry picked onto 2.1.0 before merging, looks like some changes from master are tagging along.

Done

**Comment by akhenry** [comment]
@shefalijoshi Linting error :( Looks like `expect` is imported but never used in the e2e test.

**Review by akhenry — APPROVED** [review]

**Review by unlikelyzero — CHANGES_REQUESTED** [review]
Still needs a real test

**Review by unlikelyzero — CHANGES_REQUESTED** [review]
It looks like this PR was affected when we retargeted master and picked up the 54 commits.

Can you add the e2e test stubs back? They should still be valid.

Let's sync this afternoon to figure out why codecov annotations are saying that the changes aren't covered by tests.

**Comment by shefalijoshi** [comment]
> 

I don't think the test stubs are needed anymore. We've covered clicking and resize tests as unit tests.

**Comment by akhenry** [comment]
Suspect this will fix https://github.com/nasa/openmct/issues/4206 as well

**Review by unlikelyzero — APPROVED** [review]

**Pull request merged**